### ISSUE TITLE
Error : A request is already in progress

### ISSUE THREAD

**Issue by aligassan** [dpa99c/cordova-plugin-request-location-accuracy#70]
**I'm submitting a ...**  (check one with "x"):
- [X] bug report
- [ ] feature request
- [ ] documentation issue

# Bug report
 While l am try to get permission to enable GPS without leaving app l i have error `ERROR Error: Uncaught (in promise): Object: {"code":-1,"message":"A request is already in progress"}
`
**Current behavior:**
First of all , l am using first plugin is  diagnostic and location accuracy. Every things seem working alright with diagnostic when i ask to get permission access to location . After allow to access to location i most have another ask permission for location accuracy . but i have nothing to shows . Only the error above .

**Expected behavior:**

**Steps to reproduce:**

i have error in console 

`ERROR Error: Uncaught (in promise): Object: {"code":-1,"message":"A request is already in progress"}
`

Also l am using Android Studio 'Logcat/Verbose'  to track issue and i find the following Exception 

```
2020-08-17 17:35:42.310 14271-14271/iq.tow.trucks.bgw V/Diagnostic: Received result for permissions request id=921973
2020-08-17 17:35:42.310 14271-14271/iq.tow.trucks.bgw V/Diagnostic: Authorisation for ACCESS_FINE_LOCATION is DENIED_ONCE
2020-08-17 17:35:42.311 14271-14271/iq.tow.trucks.bgw V/Diagnostic: Authorisation for ACCESS_COARSE_LOCATION is DENIED_ONCE
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art: Rejecting re-init on previously-failed class java.lang.Class<com.google.android.gms.common.SupportErrorDialogFragment>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/support/v4/app/DialogFragment;
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at com.google.android.gms.common.GoogleApiAvailability com.google.android.gms.common.GoogleApiAvailability.getInstance() ((null):1)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at void cordova.plugin.RequestLocationAccuracy.initialize(org.apache.cordova.CordovaInterface, org.apache.cordova.CordovaWebView) (RequestLocationAccuracy.java:172)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at void org.apache.cordova.CordovaPlugin.privateInitialize(java.lang.String, org.apache.cordova.CordovaInterface, org.apache.cordova.CordovaWebView, org.apache.cordova.CordovaPreferences) (CordovaPlugin.java:57)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at org.apache.cordova.CordovaPlugin org.apache.cordova.PluginManager.getPlugin(java.lang.String) (PluginManager.java:171)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at void org.apache.cordova.PluginManager.exec(java.lang.String, java.lang.String, java.lang.String, java.lang.String) (PluginManager.java:122)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at java.lang.String org.apache.cordova.CordovaBridge.jsExec(int, java.lang.String, java.lang.String, java.lang.String, java.lang.String) (CordovaBridge.java:59)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at java.lang.String org.apache.cordova.engine.SystemExposedJsApi.exec(int, java.lang.String, java.lang.String, java.lang.String, java.lang.String) (SystemExposedJsApi.java:41)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at void android.os.MessageQueue.nativePollOnce(long, int) (MessageQueue.java:-2)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at android.os.Message android.os.MessageQueue.next() (MessageQueue.java:323)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at void android.os.Looper.loop() (Looper.java:136)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at void android.os.HandlerThread.run() (HandlerThread.java:61)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.support.v4.app.DialogFragment" on path: DexPathList[[zip file "/data/app/iq.tow.trucks.bgw-1/base.apk"],nativeLibraryDirectories=[/data/app/iq.tow.trucks.bgw-1/lib/arm, /system/lib, /vendor/lib]]
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:56)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:380)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at com.google.android.gms.common.GoogleApiAvailability com.google.android.gms.common.GoogleApiAvailability.getInstance() ((null):1)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at void cordova.plugin.RequestLocationAccuracy.initialize(org.apache.cordova.CordovaInterface, org.apache.cordova.CordovaWebView) (RequestLocationAccuracy.java:172)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at void org.apache.cordova.CordovaPlugin.privateInitialize(java.lang.String, org.apache.cordova.CordovaInterface, org.apache.cordova.CordovaWebView, org.apache.cordova.CordovaPreferences) (CordovaPlugin.java:57)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at org.apache.cordova.CordovaPlugin org.apache.cordova.PluginManager.getPlugin(java.lang.String) (PluginManager.java:171)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at void org.apache.cordova.PluginManager.exec(java.lang.String, java.lang.String, java.lang.String, java.lang.String) (PluginManager.java:122)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at java.lang.String org.apache.cordova.CordovaBridge.jsExec(int, java.lang.String, java.lang.String, java.lang.String, java.lang.String) (CordovaBridge.java:59)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at java.lang.String org.apache.cordova.engine.SystemExposedJsApi.exec(int, java.lang.String, java.lang.String, java.lang.String, java.lang.String) (SystemExposedJsApi.java:41)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at void android.os.MessageQueue.nativePollOnce(long, int) (MessageQueue.java:-2)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at android.os.Message android.os.MessageQueue.next() (MessageQueue.java:323)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at void android.os.Looper.loop() (Looper.java:136)
2020-08-17 17:35:42.328 14271-14394/iq.tow.trucks.bgw I/art:     at void android.os.HandlerThread.run() (HandlerThread.java:61)
2020-08-17 17:35:42.336 14271-14271/iq.tow.trucks.bgw D/CordovaActivity: Resumed the activity.
2020-08-17 17:35:42.340 14271-14394/iq.tow.trucks.bgw I/RequestLocationAccuracy: Building GoogleApiClient
2020-08-17 17:35:42.345 14271-14394/iq.tow.trucks.bgw W/System.err: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/support/v4/util/ArrayMap;
2020-08-17 17:35:42.346 14271-14394/iq.tow.trucks.bgw W/System.err:     at com.google.android.gms.common.api.GoogleApiClient$Builder.<init>(Unknown Source)
2020-08-17 17:35:42.346 14271-14394/iq.tow.trucks.bgw W/System.err:     at cordova.plugin.RequestLocationAccuracy.buildGoogleApiClient(RequestLocationAccuracy.java:301)
2020-08-17 17:35:42.346 14271-14394/iq.tow.trucks.bgw W/System.err:     at cordova.plugin.RequestLocationAccuracy.initialize(RequestLocationAccuracy.java:173)
2020-08-17 17:35:42.346 14271-14394/iq.tow.trucks.bgw W/System.err:     at org.apache.cordova.CordovaPlugin.privateInitialize(CordovaPlugin.java:57)
2020-08-17 17:35:42.346 14271-14394/iq.tow.trucks.bgw W/System.err:     at org.apache.cordova.PluginManager.getPlugin(PluginManager.java:171)
2020-08-17 17:35:42.346 14271-14394/iq.tow.trucks.bgw W/System.err:     at org.apache.cordova.PluginManager.exec(PluginManager.java:122)
2020-08-17 17:35:42.346 14271-14394/iq.tow.trucks.bgw W/System.err:     at org.apache.cordova.CordovaBridge.jsExec(CordovaBridge.java:59)
2020-08-17 17:35:42.346 14271-14394/iq.tow.trucks.bgw W/System.err:     at org.apache.cordova.engine.SystemExposedJsApi.exec(SystemExposedJsApi.java:41)
2020-08-17 17:35:42.346 14271-14394/iq.tow.trucks.bgw W/System.err:     at android.os.MessageQueue.nativePollOnce(Native Method)
2020-08-17 17:35:42.346 14271-14394/iq.tow.trucks.bgw W/System.err:     at android.os.MessageQueue.next(MessageQueue.java:323)
2020-08-17 17:35:42.346 14271-14394/iq.tow.trucks.bgw W/System.err:     at android.os.Looper.loop(Looper.java:136)
2020-08-17 17:35:42.346 14271-14394/iq.tow.trucks.bgw W/System.err:     at android.os.HandlerThread.run(HandlerThread.java:61)
2020-08-17 17:35:42.347 14271-14394/iq.tow.trucks.bgw W/System.err: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.support.v4.util.ArrayMap" on path: DexPathList[[zip file "/data/app/iq.tow.trucks.bgw-1/base.apk"],nativeLibraryDirectories=[/data/app/iq.tow.trucks.bgw-1/lib/arm, /system/lib, /vendor/lib]]
2020-08-17 17:35:42.348 14271-14394/iq.tow.trucks.bgw W/System.err:     at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
2020-08-17 17:35:42.348 14271-14394/iq.tow.trucks.bgw W/System.err:     at java.lang.ClassLoader.loadClass(ClassLoader.java:380)
2020-08-17 17:35:42.348 14271-14394/iq.tow.trucks.bgw W/System.err:     at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
2020-08-17 17:35:42.348 14271-14394/iq.tow.trucks.bgw W/System.err: 	... 12 more
```

**Environment information**

- Cordova CLI version 
	- `cordova -v 10.0.0`
- Cordova platform version
	- `Installed platforms:
  android 8.1.0
Available platforms: 
  browser ^6.0.0
  electron ^1.0.0
  ios ^6.1.0
  osx ^6.0.0
`
- Plugins & versions installed in project (including this plugin)
    - `cordova-plugin-android-permissions 1.0.2 "Permissions"
cordova-plugin-device 2.0.2 "Device"
cordova-plugin-firebase-analytics 4.3.1 "FirebaseAnalyticsPlugin"
cordova-plugin-geolocation 4.0.2 "Geolocation"
cordova-plugin-ionic-keyboard 2.2.0 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 4.2.1 "cordova-plugin-ionic-webview"
cordova-plugin-network-information 2.0.2 "Network Information"
cordova-plugin-request-location-accuracy 2.3.0 "Request Location Accuracy"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-statusbar 2.4.2 "StatusBar"
cordova-plugin-whitelist 1.3.3 "Whitelist"
cordova-support-android-plugin 1.0.2 "cordova-support-android-plugin"
cordova-support-google-services 1.4.1 "cordova-support-google-services"
cordova.plugins.diagnostic 6.0.2 "Diagnostic"
`
- Dev machine OS and version, e.g.
    - OSX
        - `MacOs Catalina 10.15.6`
        
_Runtime issue_
- Device details
  Samsung  A72016

_Android build issue:_	
- Node JS version
    - `v12.18.3`
- Gradle version
	- `6.5.1`
- Target Android SDK version
	- `29` in `AndroidManifest.xml`
- Android SDK details
	- `sdkmanager --list | sed -e '/Available Packages/q'`
	

_If using an [Ionic Native Typescript wrapper]() for this plugin:_
- Ionic environment info
    - `   Ionic CLI                     : 6.11.0 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 5.3.1
   @angular-devkit/build-angular : 0.901.12
   @angular-devkit/schematics    : 9.1.12
   @angular/cli                  : 9.1.12
   @ionic/angular-toolkit        : 2.3.0

Cordova:

   Cordova CLI       : 10.0.0
   Cordova Platforms : not available
   Cordova Plugins   : not available

Utility:

   cordova-res : 0.15.1
   native-run  : 1.0.0

System:

   Android SDK Tools : 26.1.1 (/Users/aligassan/Library/Android/sdk)
   NodeJS            : v12.18.3 (/usr/local/bin/node)
   npm               : 6.14.6
   OS                : macOS Catalina
   Xcode             : Xcode 11.6 Build version 11E708`
- Installed Ionic Native modules and versions
    - `modules/cordova-plugin-geolocation
├─┬ @ionic-native/android-permissions@5.28.0
├─┬ @ionic-native/core@5.27.0
├─┬ @ionic-native/diagnostic@5.28.0
├─┬ @ionic-native/firebase-analytics@5.28.0
├─┬ @ionic-native/geolocation@5.27.0
├─┬ @ionic-native/location-accuracy@5.28.0
├─┬ @ionic-native/network@5.27.0
├─┬ @ionic-native/splash-screen@5.27.0
├─┬ @ionic-native/status-bar@5.27.0`

**Related code:**

```
  location() {
    this.d.requestLocationAuthorization().then((s)=>{
      console.log(s)
      return   this.la.request(this.la.REQUEST_PRIORITY_HIGH_ACCURACY).then(
        () => {
          // When GPS Turned ON call method to get Accurate location coordinates
         // this.getLocationCoordinates()
        },
        error => alert('Error requesting location permissions ' + JSON.stringify(error))
      );
    
    })

  }
```

**Console output**

<details>
<summary>console output</summary>

```

// Paste any relevant JS/native console output here

```

</details><br/><br/>

**Other information:**

# Feature request

# Documentation issue

**Comment by dpa99c** [comment]
From the Android Studio logcat, the exception indicates there's a problem with the Android Support Library in your build:

	Rejecting re-init on previously-failed class java.lang.Class<com.google.android.gms.common.SupportErrorDialogFragment>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/support/v4/app/DialogFragment;

From your list of plugins you're using `cordova.plugins.diagnostic@6.0.2` - `cordova.plugins.diagnostic@6` was a major release which migrates from the Android Support Library to AndroidX.

TL;DR: the issue is not caused by this plugin but this plugin is being affected by other issues that exist in your build.

I suggest updating your project to use `cordova-android@9` and [enabling AndroidX](https://cordova.apache.org/announcements/2020/06/29/cordova-android-9.0.0.html)


You can validate that this plugin works correctly in isolation by building and running the [example project](https://github.com/dpa99c/cordova-plugin-request-location-accuracy-example) app.
